unit-tests/src/main/scala/scala/scalanative/issues/_314.scala

```diff
@@ -0,0 +1,17 @@
+package scala.scalanative.issues
+
+object _314 extends tests.Suite {
+  test("github.com/scala-native/scala-native/issues/314") {
+    // Division by zero is undefined behavior in production mode.
+    // Optimizer can assume it never happens and remove unused result.
```

Comment:
The plan is to have two modes: one that does all the checks (development) and one that doesn't (production). You as user will be able to choose which one works better for you. https://github.com/scala-native/scala-native/issues/14